### ISSUE TITLE
deprecated kw.data and nt.names

### ISSUE THREAD

**Issue by JeffFessler** [JuliaArrays/AxisArrays.jl#200]
I'm getting a warning and an error on a nightly build:

```
 Warning: use values(kwargs) and keys(kwargs) instead of kwargs.data and kwargs.itr
│   caller = #AxisArray#7 at core.jl:243 [inlined]

...

  type DataType has no field names
  Stacktrace:
    [1] getproperty(x::Type, f::Symbol)
      @ Base ./Base.jl:37
    [2] #s11#8
      @ ~/.julia/packages/AxisArrays/IFpjG/src/core.jl:244 [inlined]
```

The error is from this apparently deprecated use of `nt.names`:
https://github.com/JuliaArrays/AxisArrays.jl/blob/9b91d546b28d96cd980e0a86d9c860c3689881d7/src/core.jl#L244

The warning is due to this apparently deprecated use of `kw.data`:
https://github.com/JuliaArrays/AxisArrays.jl/blob/9b91d546b28d96cd980e0a86d9c860c3689881d7/src/core.jl#L243

Probably related to this:
https://github.com/JuliaLang/julia/pull/39448

I'll made a PR, but I am a bit concerned about the long delays in the existing PRs, e.g., from @johnnychen94. He has even volunteered to help with maintenance here in one of his PRs but no reply in a month.  I realize the package developers are probably super busy because they have created so many terrific packages.
So how about enabling some others to help and spread the work around?  I'm also willing to help.

**Comment by johnnychen94** [comment]
It is a little bit unpleasant with this package not receiving any basic maintenance when it is not working well in Julia 1.7. I understand that people here are busy so I kindly request a write permission to this repo so that I can bring it back to track.

cc: @mbauman @timholy

**Comment by johnnychen94** [comment]
Granted kindly by @mbauman I'll switch to this package this weekend.

**Comment by johnnychen94** [comment]
closed by #201
